packages/front-end/components/Metrics/MetricForm/index.tsx

```diff
@@ -47,6 +47,13 @@ import { useCurrency } from "@/hooks/useCurrency";
 const weekAgo = new Date();
 weekAgo.setDate(weekAgo.getDate() - 7);
 
+function safeTableName(name: string) {
+  return name
+    .toLowerCase()
+    .replace(/\s+/g, "_")
+    .replace(/[^-a-zA-Z0-9_]+/g, "");
```

Comment:
So this is pretty similar to then {{snakecase}} helper.  The differences being that "some!!table" gets rendered to "some_table" instead of "sometable" like safeTableName would.  I think snakecase is good enough so I just added that and used it in the default sql templates that we are generating on the frontend.